### PR TITLE
[Dotenv] parse concatenated variable values

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #23306
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Comment by xabbuh** [comment]
The diff is best viewed at https://github.com/symfony/symfony/pull/23351/files?w=1.

**Review by chalasr — APPROVED** [review]
Works for me 👍

File: src/Symfony/Component/Dotenv/Dotenv.php [hunk 1]
```diff
@@ -168,73 +168,110 @@ private function lexValue()
...
+
+    private function lexNestedExpression()
+    {
+        ++$this->cursor;
+        $value = '';
+
+        while ("\n" !== $this->data[$this->cursor] && ')' !== $this->data[$this->cursor]) {
+            $value .= $this->data[$this->cursor];
+
+            if ('(' === $this->data[$this->cursor]) {
```
Comment (fabpot) [inline src/Symfony/Component/Dotenv/Dotenv.php]: Looks like we don't have test for that, do we?

Comment (xabbuh) [inline src/Symfony/Component/Dotenv/Dotenv.php]: https://github.com/xabbuh/symfony/blob/aaaf64ddde2aa22cb90a1c60fd902865589f5b18/src/Symfony/Component/Dotenv/Tests/DotenvTest.php#L144 and the line below trigger this part

Comment (xabbuh) [inline src/Symfony/Component/Dotenv/Dotenv.php]: But we could indeed add a test with non-matching parentheses.

**Comment by fabpot** [comment]
Thank you @xabbuh.

**Pull request merged**